### PR TITLE
validation for dots and brackets

### PR DESCRIPTION
**Pull request by mansi-amazon**
Adding validation rules to deny dots and curly brackets when creating internalusers, roles and role-mappings

### CODE REVIEW

File: public/apps/configuration/sections/internalusers/controller.js [hunk 1]
```diff
@@ -113,8 +113,13 @@ app.controller('securityEditInternalUsersController', function ($scope, $element
...
 
-        if ($scope.resourcename.indexOf("*") != -1) {
-            $scope.errorMessage = "Username must not contain '*'";
+        if ($scope.resourcename.indexOf("*") != -1 || $scope.resourcename.indexOf('.') != -1) {
+            $scope.errorMessage = "Username must not contain '*' or '.'";
+            return;
+        }
+
+        if ($scope.resourcename.indexOf("{") != -1 || $scope.resourcename.indexOf("}") != -1 != -1) {
+            $scope.errorMessage = "Username must not contain brackets";
```
Comment (dinusX) [inline public/apps/configuration/sections/internalusers/controller.js]: Can you mention hear "curly brackets {}"

File: public/apps/configuration/sections/roles/controller.js [hunk 2]
```diff
@@ -304,6 +304,12 @@ app.controller('securityEditRolesController', function ($rootScope, $scope, $ele
             event.preventDefault();
         }
 
+        // not dots in keys allowed 
+        if ($scope.resourcename.indexOf('.') != -1 || $scope.resourcename.indexOf("{") != -1 || $scope.resourcename.indexOf("}") != -1) {   
+            $scope.errorMessage = 'Please do not use dots or brackets in the role name.';
```
Comment (dinusX) [inline public/apps/configuration/sections/roles/controller.js]: Similarly, mention "curly brackets {} " for other messages.

File: public/apps/configuration/sections/internalusers/controller.js (hunk 1 above)
Comment (mansi-amazon) [reply public/apps/configuration/sections/internalusers/controller.js]: updated

File: public/apps/configuration/sections/roles/controller.js (hunk 2 above)
Comment (mansi-amazon) [reply public/apps/configuration/sections/roles/controller.js]: updated

File: public/apps/configuration/sections/internalusers/controller.js [hunk 3]
```diff
@@ -113,8 +113,13 @@ app.controller('securityEditInternalUsersController', function ($scope, $element
...
         }
 
-        if ($scope.resourcename.indexOf("*") != -1) {
-            $scope.errorMessage = "Username must not contain '*'";
+        if ($scope.resourcename.indexOf("*") != -1 || $scope.resourcename.indexOf('.') != -1) {
+            $scope.errorMessage = "Username must not contain '*' or '.'";
+            return;
+        }
+
+        if ($scope.resourcename.indexOf("{") != -1 || $scope.resourcename.indexOf("}") != -1 != -1) {
```
Comment (hardik-k-shah) [inline public/apps/configuration/sections/internalusers/controller.js]: we should merge this with above if condition for better customer experience.
Also message too.

File: public/apps/configuration/sections/roles/controller.js [hunk 4]
```diff
@@ -304,6 +304,12 @@ app.controller('securityEditRolesController', function ($rootScope, $scope, $ele
             event.preventDefault();
         }
 
+        // not dots in keys allowed 
+        if ($scope.resourcename.indexOf('.') != -1 || $scope.resourcename.indexOf("{") != -1 || $scope.resourcename.indexOf("}") != -1) {   
+            $scope.errorMessage = 'Please do not use dots or curly brackets in the role name.';
```
Comment (hardik-k-shah) [inline public/apps/configuration/sections/roles/controller.js]: are we allowing '*' for role name?

File: public/apps/configuration/sections/internalusers/controller.js (hunk 3 above)
Comment (mansi-amazon) [reply public/apps/configuration/sections/internalusers/controller.js]: merged

Comment (mansi-amazon) [reply public/apps/configuration/sections/internalusers/controller.js]: fixed type also

File: public/apps/configuration/sections/roles/controller.js (hunk 4 above)
Comment (mansi-amazon) [reply public/apps/configuration/sections/roles/controller.js]: checking for '*' also now

**Pull request merged**